nifi-registry-core/nifi-registry-web-api/src/test/resources/keys/README.md

```diff
@@ -12,36 +12,217 @@
+The CA and private keys (including their protection passwords) have been published on the Internet, so they should never be trusted.
 
-    # copy server's key/trust stores
-    mkdir keys
-    cp localhost/keystore.jks keys/localhost-ks.jks
-    cp localhost/truststore.jks keys/localhost-ts.jks
+***  
 
-    # create a Java Key Store (JKS) from the client key
-    keytool -importkeystore -destkeystore keys/client-ks.jks -deststorepass clientKeystorePassword -destkeypass u1Pass -srckeystore CN=user1_OU=nifi.p12 -srcstorepass u1Pass -srcstoretype PKCS12
+## Directory Contents
 
+### Certificate Authority (CA)
 
-You should now have a directory with the following contents:
+| Hostname / DN | File | Description | Format | Password |
+| --- | --- | --- | --- | --- |
+| - | ca-cert.pem | CA public cert | PEM (unencrypted) | N/A |
+| - | ca-key.pem | CA private (signing) key | PEM | password |
+| - | ca-ts.jks | CA cert truststore (shared by clients and servers) | JKS | password |
+| - | ca-ts.p12 | CA cert truststore (shared by clients and servers) | PKCS12 | password |
+| registry, localhost | registry-cert.pem | NiFi Registry server public cert | PEM (unencrypted) | N/A |
+| registry, localhost | registry-key.pem | NiFi Registry server private key | PEM | password |
+| registry, localhost | registry-ks.jks | NiFi Registry server key/cert keystore | JKS | password |
+| registry, localhost | registry-ks.p12 | NiFi Registry server key/cert keystore | PKCS12 | password |
+| CN=user1, OU=nifi | user1-cert.pem | client (user="user1") public cert | PEM (unencrypted) | N/A |
+| CN=user1, OU=nifi | user1-key.pem | client (user="user1") private key | PEM | password |
+| CN=user1, OU=nifi | user1-ks.jks | client (user="user1") key/cert keystore | JKS | password |
+| CN=user1, OU=nifi | user1-ks.p12 | client (user="user1") key/cert keystore | PKCS12 | password |
 
-    keys/
-     +-- client-ks.jks      # client keystore: keystorePass=clientKeystorePassword, keyPass=u1Pass
-     +-- localhost-ks.jks   # server keystore: keystorePass=localhostKeystorePassword, keyPass=localhostKeystorePassword
-     +-- localhost-ts.jks   # server/client truststore (contains CA): truststorePass=localhostTruststorePassword
+## Generating Additional Test Keys/Certs
 
-Copy these files to the test/resources/keys/ directory.
+If we need to add a service to our test environment that requires a cert signed by the same CA, here are the steps for generating additional keys for thsi directory that are signed by the same CA key.
 
+Requirements:
+
+- docker
+- keytool (included with Java)
+- openssl (included/available on most platforms)
+
+If you do not have docker, you can alternatively use the nifi-toolkit binary which has a dependency on Java 1.8 and is available for download from http://nifi.apache.org. 
+
+### New Service Keys
+
+The steps for generating a new *service* key/cert pair are (using `proxy` as the example service):
+
+```
+# make working directory
+rm -rf /tmp/test-keys
```

Comment:
Is removing the directory necessary? We should include a note explaining what will happen here. While it's obvious to most users, some may have generated keys in that directory that they are still using, etc.

Reply:
Yeah I was on the fence about that. In the end I was more worried that they would be copying old keys to the final output directory mistakenly. could be convinced either way...

Reply:
Understood. I guess it's ok because everything can be recreated, but I just worry about providing users with copy/paste commands that take destructive action in the event those users don't understand the consequences. Theoretically, one would never have to worry about that in a security context, setting up keystores and certificates, but obviously it happens. 

Could provide a script which named the subdirectory using something like `$(date +"%T")` so they are unique, but it might make it harder for users to know which subdirectory to copy from.

Reply:
@alopresto the latest commit addresses this. Feel free to squash commits on merge and discard the fix commit messages.